### DISCUSSION TITLE
Email field inconsistent in database schema and types

### DISCUSSION THREAD

**Discussion by medihack** [nextauthjs/next-auth#6216] [Help]
The `email` field seems to be a bit inconsistent in database schemas and types.

- It's required in [type `AdapterUser`](https://github.com/nextauthjs/next-auth/blob/2e52a89c52676a39b5f0764b27fe7199967d4d1c/packages/core/src/adapters.ts#L78).
- It's optional in the [schema documentation of Prisma](https://next-auth.js.org/adapters/prisma#create-the-prisma-schema).
- It's required in a [Prisma schema I found in the source code](https://github.com/nextauthjs/next-auth/blob/2e52a89c52676a39b5f0764b27fe7199967d4d1c/packages/adapter-prisma/prisma/schema.prisma#L13).

I wonder now, can I assume that an `email` is always present or not?

**Comment by uncvrd** [comment]
I'd like to know because I noticed this discrepancy as well. It appears some providers do not return an email which is why I believe the prisma adapter email is optional. For example, twitter does not support email apparently and returns null https://github.com/nextauthjs/next-auth/blob/3dfdca1bce1d1f2198cbdd6d08b0a0073c5aeba3/packages/core/src/providers/twitter.ts#L176

So I guess it's up to the developer to choose the providers with a hard email requirement
